src/components/VideoListing/MoreOptionsModal.js

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useRef, useEffect } from "react";
+import "./VideoListing.css";
+import { RiShareForwardLine } from "react-icons/ri";
+
+export const MoreOptionsModal = () => {
+  const ref = useRef();
+  const [showOptionsModal, setShowOptionsModal] = useState(false);
+
+  useEffect(() => {
+    const checkIfClickedOutside = e => {
+      if (showOptionsModal && ref.current && !ref.current.contains(e.target)) {
+        setShowOptionsModal(false);
+      }
+    };
+    document.addEventListener("mousedown", checkIfClickedOutside);
+
+    return () => {
+      // Cleanup the event listener
+      document.removeEventListener("mousedown", checkIfClickedOutside);
+    };
+  }, [showOptionsModal]);
```

Comment:
Question: Is it the only way to toggle model as it is in vanilla js in React. we should avoid vanilla js in react right?

Reply:
I've tried to do it with only react but couldn't get the desired result. And many blogs have suggested this way. So we might have to use some vanilla js for specific use cases.